### PR TITLE
docs(sass): add public and private sass API annotations

### LINKED ISSUE #1856: Document public Sass API for v10
In the past, we've exposed our Sass API in general for folks to use. As we are guarding against breaking changes, this has been causing troubles when it comes to build breakage. As a result, we should document our public Sass API in `src/globals/scss/README.md` (at least initially) as a point of reference for people to understand what we support from the Sass perspective.

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #1856 

**Reviewing**
Let me know if there's any description or example that is incorrect.

### CODE REVIEW

File: src/globals/scss/_feature-flags.scss [hunk 1]
```diff
@@ -9,10 +9,16 @@
 // 🎌 Feature Flags
 //-------------------------
 
-// Initialize our feature flag map with default values
+/// @access public
+/// @type Map | Bool
```
Comment (asudoh) [inline src/globals/scss/_feature-flags.scss]: Any scenario where `$feature-flags` becomes a bool...?

File: src/globals/scss/_layer.scss [hunk 2]
```diff
@@ -30,6 +34,10 @@ $box-shadow--temporary-nav: 0 8px 16px 0 $box-shadow;
 $box-shadow--pop-out: 0 12px 24px 0 $box-shadow;
 
 // Layer box-shadow map
+/// @access private
+/// @type Map | Value
```
Comment (asudoh) [inline src/globals/scss/_layer.scss]: Any scenario where `$layer-shadows` becomes a non-map value...?

File: src/globals/scss/_layer.scss [hunk 3]
```diff
@@ -41,6 +49,10 @@ $layer-shadows: (
 );
 
 // Layer mixin
+/// @access public
+/// @param {Map} $layer - a value from the $layer-shadows map
```
Comment (asudoh) [inline src/globals/scss/_layer.scss]: I think `$layer` is a key of `$layer-shadows`, thus a string?

File: src/globals/scss/_layout.scss [hunk 4]
```diff
@@ -23,12 +26,24 @@ $breakpoints: (
...
   'mobile': 3%,
   'xs': 5%,
 );
 
+//-------------------------------------
+// @function: padding($value)
+//-------------------------------------
 /// @deprecated (For v10) Was used primary for older grid
+/// @group $padding map
+/// @param {Map} $value - mobile or xs
```
Comment (asudoh) [inline src/globals/scss/_layout.scss]: I think `$value` is a key of `$padding`, thus a string?

File: src/globals/scss/_layout.scss [hunk 5]
```diff
@@ -40,13 +55,14 @@ $padding: (
...
 //-------------------------------------
-// usage:
-// @include breakpoint(bp--md--major) {
-//    - styles here --
-// }
-//-------------------------------------
-
+/// @access public
 /// @deprecated (For v10) Use `@include carbon--breakpoint()`
+/// @param {Map} $size - from $breakpoints map (see line 16)
```
Comment (asudoh) [inline src/globals/scss/_layout.scss]: I think `$size` is a key of `$breakpoints`, thus a string?

File: src/globals/scss/_layout.scss [hunk 6]
```diff
@@ -65,7 +81,14 @@ $padding: (
   }
 }
 
+//-------------------------------------
+// @mixin: max-breakpoint($size)
+//-------------------------------------
+/// @access public
 /// @deprecated (For v10) Use `@include carbon--breakpoint-down()`
+/// @param {Map} $size - from $breakpoints map (see line 16)
```
Comment (asudoh) [inline src/globals/scss/_layout.scss]: I think `$size` is a key of `$breakpoints`, thus a string?

File: src/globals/scss/_typography.scss [hunk 7]
```diff
@@ -37,7 +47,11 @@ $typescale-map: (
   'p': 1rem,
 );
 
+/// @access public
 /// @deprecated (For v10) Use `@include carbon--type-scale()`
+/// @group typography
+/// @param {Map} $size - value of type size
```
Comment (asudoh) [inline src/globals/scss/_typography.scss]: I think `$size` is a key of `$typescale-map`, thus a string?

File: src/globals/scss/_typography.scss [hunk 8]
```diff
@@ -122,6 +157,9 @@ $typescale-map: (
   }
 }
 
+///@access public
```
Comment (asudoh) [inline src/globals/scss/_typography.scss]: Formatting nit:

```suggestion
/// @access public
```

File: src/globals/scss/_typography.scss [hunk 9]
```diff
@@ -136,7 +174,12 @@ $font-size-map: (
   '11': 0.6875rem,
 );
 
+/// @access public
 /// @deprecated (For v10) Use `@include carbon--type-scale()`
+/// @param {Map} $size value from $font-size-map
```
Comment (asudoh) [inline src/globals/scss/_typography.scss]: I think `$size` is a key of `$font-size-map`, thus a string?

**Review by asudoh — COMMENTED** [review]
Great work - Thanks a lot @jnm2377 for doing this! In addition to my comments on code - I see changes in `_css--body.scss` that are not related to your work. Would you be able to back them out?

File: src/globals/scss/_css--body.scss [hunk 10]
```diff
@@ -21,22 +22,8 @@
   }
 }
 
-@mixin css-body--x {
```
Comment (joshblack) [inline src/globals/scss/_css--body.scss]: Any reason for dropping the `x` variant here?

**Comment by asudoh** [comment]
@jnm2377 Added several commits for bringing back `@mixin css--body` and fixing several SassDoc annotations. Hoping it doesn't break any of your intents, but please don't hesitate to speak up otherwise. Thanks!

**Review by asudoh — APPROVED** [review]
And thank you for your great work here @jnm2377!

**Review by joshblack — CHANGES_REQUESTED** [review]
Ignore me! Went ahead and pushed up some slight tweaks to help get this in!

**Review by joshblack — APPROVED** [review]

**Comment by asudoh** [comment]
Need to decide if `@mixin css--body` should be public or private, and the name of the mix-in if we are making it public, but I don't want to block this PR on that. Thanks again @jnm2377 for all your work and @joshblack for the tweaks!

**Pull request merged**